### PR TITLE
Helm with Ingress included

### PR DESCRIPTION
**Pull request by oldgiova**
# External Contributor Checklist

🚨 Please review the [guidelines for contributing](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md) to this repository.

- [x] Make sure that all commits follow the conventional commit [specification](https://github.com/mendersoftware/mendertesting/blob/master/commitlint/grammar.md) for the Mender project.

The majority of our contributions are fixes, which means your commit should have
the form below:

```
fix: Added documentation for the Ingress included in the Helm Chart

Changelog: Added documentation for the Ingress included in the Helm Chart
Ticket: MC-6912
```

- [x] Make sure that all commits are signed with [`git --signoff`](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work). Also note that the signoff author must match the author of the commit.

### Description

Added documentation for the Ingress included in the Helm Chart

**Note:** this has to be cherry picked to the 3.6 or later releases.

### CODE REVIEW

File: 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md [hunk 1]
```diff
@@ -53,7 +53,9 @@ helm repo add mender https://charts.mender.io
 helm repo update
 ```
 
-You can now install the Mender Server running:
+You can now install the Mender Server. You also have to decide how to expose
+the Mender API Gateway service. This Helm Chart includes a configurable Ingress resource
+for your convenience. Some sample Ingress values are detailed on the later section.
```
Comment (tranchitella) [inline 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md:58]:
```suggestion
for your convenience. Some sample Ingress values are detailed in the later section.
```

File: 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md [hunk 2]
```diff
@@ -240,37 +242,92 @@ an [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) o
...
 to expose the Mender Server outside the Kubernetes cluster.
 
-For example, to expose the Mender Server with an Ingress, run:
-
+**Required configuration**:
+* Redirection from HTTP to HTTPS on the Ingress is required
+* If the health check is required, the internal path is `/ui`
+* Either a valid certificate for the `${MENDER_SERVER_DOMAIN}` must exists as a secret named `mender-ingress-tls`, or
+  you already have configured the [Cert-manager](../01.Kubernetes/docs.md#installation-of-cert-manager-optional) resource
+* The Mender helm chart is deployed with `api_gateway.env.SSL=false` option:
```
Comment (tranchitella) [inline 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md:250]:
```suggestion
* The Mender Helm chart is deployed with the `api_gateway.env.SSL=false` option:
```

File: 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md [hunk 3]
```diff
@@ -240,37 +242,92 @@ an [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) o
...
+  ```
+  api_gateway:
+    env:
+      SSL: false
+  ```
+
+<!--AUTOVERSION: "Starting from the Mender Helm Chart version %"/ignore -->
+You can refer to your Infrastructure Provider's documentation for how to create an Ingress resource.
```
Comment (tranchitella) [inline 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md:258]:
```suggestion
You can refer to your Infrastructure Provider's documentation for creating an Ingress resource.
```

File: 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md [hunk 4]
```diff
@@ -240,37 +242,92 @@ an [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) o
...
+Starting from the Mender Helm Chart version 5.1.0, a sample Ingress resource is included.
+
+For example, here's an Ingress for the AWS EKS Provider:
+
+<!--AUTOVERSION: "cat <<-EOF >> mender-%.yml"/ignore -->
```
Comment (tranchitella) [inline 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md:263]:
```suggestion
<!--AUTOVERSION: "cat <<-EOF >> mender-%.yml"/integration -->
```

File: 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md [hunk 5]
```diff
@@ -240,37 +242,92 @@ an [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) o
...
 ```bash
-cat >mender-ingress.yml <<EOF
-apiVersion: networking.k8s.io/v1
-kind: Ingress
-metadata:
-  name: mender-ingress
+cat <<-EOF >> mender-3.6.0-build3.yml
```
Comment (tranchitella) [inline 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md:265]:
```suggestion
cat <<-EOF >> mender-3.6.1.yml
```

File: 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md [hunk 6]
```diff
@@ -240,37 +242,92 @@ an [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) o
...
+  # this secret must exists or it can be created from a working cert-manager instance
+   - secretName: mender-ingress-tls
+     hosts:
+       - ${MENDER_SERVER_DOMAIN}
+EOF
+```
+
+For example, here's an Ingress for Azure AKS Provider:
+
+<!--AUTOVERSION: "cat <<-EOF >> mender-%.yml"/ignore -->
```
Comment (tranchitella) [inline 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md:298]:
```suggestion
<!--AUTOVERSION: "cat <<-EOF >> mender-%.yml"/integration -->
```

File: 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md [hunk 7]
```diff
@@ -240,37 +242,92 @@ an [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) o
...
+```bash
+cat <<-EOF >> mender-3.6.0-build3.yml
```
Comment (tranchitella) [inline 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md:300]:
```suggestion
cat <<-EOF >> mender-3.6.1.yml
```

File: 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md [hunk 8]
```diff
@@ -240,37 +242,92 @@ an [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) o
...
+  # this secret must exists or it can be created from a working cert-manager instance
+   - secretName: mender-ingress-tls
+     hosts:
+       - ${MENDER_SERVER_DOMAIN}
 EOF
+```
 
-kubectl apply -f mender-ingress.yml
+You can now update the Helm Chart with the included ingress:
+<!--AUTOVERSION: "helm upgrade --install mender mender/mender -f mender-%.yml"/ignore -->
```
Comment (tranchitella) [inline 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md:321]:
```suggestion
<!--AUTOVERSION: "helm upgrade --install mender mender/mender -f mender-%.yml"/integration -->
```

File: 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md [hunk 9]
```diff
@@ -240,37 +242,92 @@ an [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) o
...
+```bash
+helm upgrade --install mender mender/mender -f mender-3.6.0-build3.yml
```
Comment (tranchitella) [inline 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md:323]:
```suggestion
helm upgrade --install mender mender/mender -f mender-3.6.1.yml
```

**Review by tranchitella — CHANGES_REQUESTED** [review]
Please use the autoversioning.

File: 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md (hunk 6 above)
Comment (oldgiova) [reply 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md:307]: This is an Azure example, won't work on the EKS cluster for the Integration tests.

**Comment by oldgiova** [comment]
> Please use the autoversioning.

I'm a bit unsure...

My original idea was to provide some examples/guidance for EKS and AKS setups. Some key points are missing (secret, certificate management...) because it could vary depending on the implementation.

Maintaining this in the CI pipeline requires more effort because we have to push our own secrets and ingress conf in the Autoversion, and also actually testing the Ingress (simply applying it is useless: it could fail after creation).

Also considering that the `ingress` is disabled by default.

I'm going to add the autoversion in a separate commit and let's see how it goes on the actual integration tests.

**Review by tranchitella — COMMENTED** [review]
I don't known own anymore, therefore I'm not the one supposed to review this stuff.

**Review by tranchitella — APPROVED** [review]
Marking "Approved" only to remove the "Request changes" status.

**Review by MuchoLucho — APPROVED** [review]
It does make sense

File: 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md [hunk 10]
```diff
@@ -249,35 +251,124 @@ an [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) o
 [Load Balancer](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer)
 to expose the Mender Server outside the Kubernetes cluster.
 
-For example, to expose the Mender Server with an Ingress, run:
-
+**Required configuration**:
```
Comment (merlin-northern) [inline 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md:254]: I am having doubts about the number of ** second level has two, right?

File: 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md [hunk 11]
```diff
@@ -249,35 +251,124 @@ an [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) o
...
+* Redirection from HTTP to HTTPS on the Ingress is required
+* If the health check is required, the internal path is `/ui`
```
Comment (merlin-northern) [inline 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md:256]: tons of passive voice, the test may stop you.

File: 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md [hunk 12]
```diff
@@ -249,35 +251,124 @@ an [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) o
...
+For example, here's an Ingress for the AWS EKS Provider:
+
+<!--AUTOVERSION: "cat <<-EOF >> mender-%.yml"/integration -->
 ```bash
-cat >mender-ingress.yml <<EOF
-apiVersion: networking.k8s.io/v1
-kind: Ingress
-metadata:
-  name: mender-ingress
+cat <<-EOF >> mender-3.6.3.yml
```
Comment (merlin-northern) [inline 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md:274]: this is `>>` (append) on purpose?

**Review by merlin-northern — APPROVED** [review]
dropped some comments take them or leave them ;-) but like them!

File: 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md (hunk 10 above)
Comment (oldgiova) [reply 07.Server-installation/04.Production-installation-with-kubernetes/03.Mender-server/docs.md:254]: Here it's bold: 

**Required configuration**:
* Redirection from HTTP to HTTPS on the Ingress is required
* If the health check is required, the internal path is `/ui`

**Comment by oldgiova** [comment]
That version was too outdated: created a new one here: https://github.com/mendersoftware/mender-docs/pull/2173

**Pull request closed without merging**